Allow admins to choose not exchanging for specific SSO token

NobodysNightmare opened the pull request:
This is intended for simple setups, where the token returned by the IDP can be immediately used for authentication at a storage. It is also required for integrating with Nextcloud as an IDP.

## Ticket
https://community.openproject.org/projects/cross-application-user-integration-stream/work_packages/61623

## What are you trying to accomplish?
It should be possible for admins to use the admin token obtained from the IDP directly for authentication at the storage.

## Screenshots

![image](https://github.com/user-attachments/assets/1a9344c2-585e-423b-a59e-ef4dedd87309)

## Merge checklist

- [x] Added/updated tests
- [x] Tested firefox/chromium

NobodysNightmare committed:
Allow admins to choose not exchanging for specific SSO token

This is intended for simple setups, where the token returned by
the IDP can be immediately used for authentication at a storage.

NobodysNightmare committed:
Replace the term nextcloud audience with storage audience

The components we are starting to build should be generally usable
for different storages. Before we sink more cost into overly
specific naming, we should generalize to a name that can be used
anywhere, if we want to.

Left out components that would require renaming files for now, to reduce
the conflict potential a bit.

NobodysNightmare committed:
Add feature tests around storage audience component

Testing correct showing and hiding behaviour as well as
persistence of the corresponding selection.

ba1ash commented on frontend/src/stimulus/controllers/dynamic/storages/storage-audience.controller.ts:
```diff
@@ -0,0 +1,58 @@
+  }
+
+  showAudienceInput() {
+    this.audienceInputTarget.value = '';
```
:yellow_circle: 
Setting value to be empty on every show leads to user loosing its input which can be inconvenient. For example, in `Automatically managed folder` section of the same page password is not removed when user clicks on checkbox.
Therefore I suggest not clearing input value, but then `__op-idp__` can appear in the input field. I'm not sure if that is critical.

NobodysNightmare replied:
> but then __op-idp__ can appear in the input field

This is exactly why I added the clearing, to prevent this "magic" audience from leaking through.

I understand the general inconvenience of losing an input, on the other hand we could at best prevent losing it when you click around in the form. As soon as you save, your input is lost anyways, because the radio button is just a way to set the same data.

Let me try something...

NobodysNightmare replied:
Thanks for nudging me over. I found a solution that is much cleaner, because we never even send the internal identifier to the browser.

Feel free to have a look at my latest commit.

ba1ash replied:
Looks nice to me! Thanks for taking care of it.

ba1ash reviewed (approved):
Well done!
I have one UX related remark.

NobodysNightmare committed:
Do not clear audience in JS

This made the UX less than optimal, because users "playing"
with the radio buttons could lose their input.

The problem we wanted to solve was preventing that the magic
internal audience name for the IDP audience would leak into the
UI. This is now prevented earlier, by making sure it's never
even filled into the input field. Thus the frontend side code
now doesn't have to care about this at all.
